### ISSUE TITLE
methods should not panic

### ISSUE THREAD

**Issue by anna-is-cute** [Feliix42/anybar-rs#1]
Instead of panicking, which makes it near-impossible to do error handling, return a Result.

``` rust
fn set_color(&self, color: Color) -> Result<(), ErrorType>
```

If the UDP socket couldn't be bound, return an Error instead of panicking. If the request was sent successfully, return `Ok(())` or some other meaningful result.

**Comment by anna-is-cute** [comment]
As an example, this is what I have to do to use this library in any program that should be stable at all:

``` rust
let result = panic::catch_unwind(|| {
  let mut anybar = Anybar::default();
  anybar.set_color(color);
});
```

**Comment by Feliix42** [comment]
By the time I designed this it seemed like a good choice but only because my use cases back then were very limited. 
Now I agree with you and am going to replace this with a `Result`.

**Comment by Feliix42** [comment]
Should be fixed via f90baa3.
